### DISCUSSION TITLE
In-memory buffer

### DISCUSSION THREAD

**Discussion by matthughes** [outr/scribe#394] [Q&A] [answered]
I see that the async logger has the concept of an in-memory buffer where presumably it buffers log messages up to the buffer size and then writes it out to some sink.

Is there any feature where you can have that in-memory circular buffer and then programmatically dump the buffer to disk on command?  In a system where you didn't want the overhead of logging or you worried about disk space, the idea would be the logging is still there but you only dump it to disk in case of some sort of diagnostic report.

**Comment by darkfrog26** [comment]
That should be pretty trivial to add even as an extension library. If you think it would be valuable I don't have any issues adding it to the library directly though. Would you be interested in putting together a PR for it?

**Comment by darkfrog26** [comment, marked as answer]
There is a new `Caching LogHandler` that has been added in `3.11.0` that should resolve this.
